File: CaesarCipher.java
/*
* Case Sensitive Caesar Cipher Using Java
* Author : wh0am1
* GitHub : https://github.com/wh0th3h3llam1
*/

/* Case Sensitive Caesar Cipher */


import java.lang.*;
import java.util.Scanner;
import java.io.*;


class CaesarCipher
{
	public static void main(String[] args) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int ch;
		String Lin = "Linux";
		String Win = "Windows 10";
		while(true)
		{
			if(System.getProperty("os.name").equals(Lin))
			{
				System.out.print("\033[H\033[2J");
			}
			if(System.getProperty("os.name").equals(Win))
			{
				new ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor();
			}
			System.out.flush();
			System.out.println("\t\t\t\t\tCasear Cipher");
			System.out.println();
			System.out.println("\t1. Encrypt A String");
			System.out.println("\t2. Decrypt A String");
			System.out.println("\t0. Exit");
			System.out.println();
			System.out.print("\tEnter Your Choice : ");

			ch = sc.nextInt();

			switch(ch)
			{
				case 1 :
					Encrypt en = new Encrypt();
					try
					{
						en.getData();
					}catch(Exception e){}
					String enc = en.getEncryptedString();
					System.out.println("\n\tEncrypted String : " + enc);
					System.out.println();
					break;
				case 2 :
					Decrypt de = new Decrypt();
					try
					{
						de.getData();
					}catch(Exception e){}
					String dec = de.getDecryptedString();
					System.out.println("\n\tDecrypted String : " + dec);
					System.out.println();
					break;
				case 0 :
					System.exit(0);
				default :
					System.out.println("\n\tInvalid Choice.....Please Try Again");
					System.out.println();
					break;
			}
			System.in.read();
		}
	}
}


class Encrypt
{
	String str="";
	int shift=0;

	void getData() throws Exception
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("\n\tEnter A String : ");
		str = br.readLine();
		System.out.print("\tEnter Shift Value : ");
		shift = Integer.parseInt(br.readLine());
	}

	String getEncryptedString()
	{
		String s = "";
		for(int i = 0; i < str.length(); i++)
		{
			char c = 'a';

			// Character is Between A-Z
			if((str.charAt(i)) >= 65 && (str.charAt(i)) <= 90)
			{
				if(((str.charAt(i)) + shift) > 90)
				{
					c = (char)(((str.charAt(i)) + shift - 65) % 26 + 65);
				}
				else
				{
					c = (char)((str.charAt(i)) + shift);
				}
			}

			// Character is Between a-z
			else if((str.charAt(i)) >= 97 && (str.charAt(i)) <= 122)
			{
				if(((str.charAt(i)) + shift) > 122)
				{
					c = (char)(((str.charAt(i)) + shift - 97) % 26 + 97);
				}
				else
				{
					c = (char)((str.charAt(i)) + shift);
				}
			}

			// Character is Between 0-9
			else if((str.charAt(i)) >= 48 && (str.charAt(i)) <= 57)
			{
				if(((str.charAt(i)) + shift) > 57)
				{
					c = (char)(((str.charAt(i)) + shift - 48) % 10 + 48);
				}
				else
				{
					c = (char)((str.charAt(i)) + shift);
				}
			}

			// Character is a Space ' '
			else if(str.charAt(i) == 32)
			{
				c = (char)(str.charAt(i));
			}

			// Character is a special character
			else
			{
				c = (char)(str.charAt(i));
			}
			String ci = Character.toString(c);
			s += ci;
		}
		return s;
	}
}

class Decrypt
{
	String str="";
	int shift=0;

	void getData() throws Exception
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		System.out.print("\n\tEnter A String : ");
		str = br.readLine();
		System.out.print("\tEnter Shift Value : ");
		shift = Integer.parseInt(br.readLine());
	}

	String getDecryptedString()
	{
		String s = "";
		for(int i = 0; i < str.length(); i++)
		{
			char c = 'a';

			// Character is Between A-Z
			if((str.charAt(i)) >= 65 && (str.charAt(i)) <= 90)
			{
				if(((str.charAt(i)) - shift) < 65)
				{
					c = (char)(((str.charAt(i)) - shift - 65 + 26) % 26 + 65);
				}
				else
				{
					c = (char)((str.charAt(i)) - shift);
				}
			}

			// Character is Between a-z
			else if((str.charAt(i)) >= 97 && (str.charAt(i)) <= 122)
			{
				if(((str.charAt(i)) - shift) < 97)
				{
					c = (char)(((str.charAt(i)) - shift - 97 + 26) % 26 + 97);
				}
				else
				{
					c = (char)((str.charAt(i)) - shift);
				}
			}

			// Character is Between 0-9
			else if((str.charAt(i)) >= 48 && (str.charAt(i)) <= 57)
			{
				if(((str.charAt(i)) - shift) < 48)
				{
					c = (char)(((str.charAt(i)) - shift - 48 + 10) % 10 + 48);
				}
				else
				{
					c = (char)((str.charAt(i)) - shift);
				}
			}

			// Character is a Space ' '
			else if(str.charAt(i) == 32)
			{
				c = (char)(str.charAt(i));
			}

			// Character is a special character
			else
			{
				c = (char)(str.charAt(i));
			}
			String ci = Character.toString(c);
			s += ci;
		}
		return s;
	}
}

/* wh0am1 */

File: CaesarCipher.py
# Caesar Cipher (Additive Cipher) using Python
# Case Sensitive
# Author : wh0am1
# GitHub : https://github.com/wh0th3h3llam1


import os
import platform

def encrypt(strng, key):

	enc = []

	for i in range(len(strng)):
		if ord(strng[i]) in range(97, 123):
			# Small Letter

			t = (ord(strng[i]) - 97 + key) % 26 + 97
			enc.append(chr(t))

		elif ord(strng[i]) in range(65, 91):
			# Capital Letter

			t = (ord(strng[i]) - 65 + key) % 26 + 65

			enc.append(chr(t))

		elif ord(strng[i]) in range(48, 58):
			# Numbers

			t = (ord(strng[i]) - 48 + key) % 10 + 48
			enc.append(chr(t))

		else:
			enc.append(strng[i])
	encrypted = ""
	print("Encrypted String : " + encrypted.join(enc))
	return encrypted


def decrypt(strng, key):
	dec = []

	for i in range(len(strng)):
		if ord(strng[i]) in range(97, 123):
			# Small Letter

			t = (ord(strng[i]) - 97 - key) % 26 + 97
			dec.append(chr(t))

		elif ord(strng[i]) in range(65, 91):
			# Capital Letter

			t = (ord(strng[i]) - 65 - key) % 26 + 65
			dec.append(chr(t))

		elif ord(strng[i]) in range(48, 58):
			# Numbers

			t = (ord(strng[i]) - 48 - key) % 10 + 48
			dec.append(chr(t))

		else:
			dec.append(strng[i])
	decrypted = ""
	print("Decrypted String : " + decrypted.join(dec))
	return decrypted


def brute_force(strng):
	key = 26
	print("Brute Forcing the String.....")

	for k in range(key):
		print("Key : " + "{:02d}".format(k) + " | Decrypted String : ", end="")
		decrypt(strng, k)


def main():
	while True:
		if platform.system() == 'Windows':
			os.system('cls')
		else:
			os.system('clear')

		print("****************************************************")
		print("\t\tCAESAR CIPHER (ADDITIVE CIPHER)")
		print("****************************************************")

		print("1. Encrypt")
		print("2. Decrypt")
		print("3. Brute-Force")
		print("0. Exit")
		print("Enter Your Choice : ", end="")
		choice = int(input())

		print("----------------------------------------------------")

		if choice == 1:
			strng = input("Enter String to Encrypt : ")
			key = int(input("Enter Key : "))
			key = key % 26
			print("----------------------------------------------------")
			encrypt(strng, key)

		elif choice == 2:
			strng = input("Enter String to Decrypt : ")
			key = int(input("Enter Key : "))
			key = key % 26
			print("----------------------------------------------------")
			decrypt(strng, key)

		elif choice == 3:
			strng = input("Enter String to Brute-Force : ")
			print("----------------------------------------------------")
			brute_force(strng)

		elif choice == 0:
			print("BYE...")
			print("----------------------------------------------------")
			exit(0)

		else:
			# print("Invalid Choice, Try Again.")
			pass

		x = input()



if __name__ == '__main__':
	main()
